gh-pages/_posts/how-to/2020-08-17-fixate-folders-with-a-custom-cc-json.md

```diff
@@ -0,0 +1,101 @@
+The layout varies depending on the area of each node. Adding a new file to a folder might increase the area of the parent folder,
+so that the algorithm decides that the parent-folder needs to be relocated. Another way to follow this problem is by changing the area-metric.
+Notice, that most folders will not be located at the same position as before.
+
+# Fixate folders
+
+In order to prevent folders from changing locations, we introduced a new attribute to the `.cc.json` called `fixedPosition`.
+This attribute is not auto-generated during the analysis and has to be defined manually by editing the `.cc.json`.
+
+Setting this property is restricted to the top level folders and won't have any effect on sub-folders (folders in folders).
+
+# API
+
+```json
+ "fixedPosition": {
+    "left": 60,
+    "top:": 40,
+    "width": 35,
+    "height": 55
+ }
+```
+
+The property values must be numbers in the range between 0 and 100. They represent the size of the value in percent.
+
+## Example
+
+```json
+{
+	"projectName": "example-project",
+	"apiVersion": "1.2",
+	"nodes": [
+		{
+			"name": "root",
+			"type": "Folder",
+			"attributes": {},
+			"children": [
+				{
+					"name": "folder_1",
+					"type": "Folder",
+					"attributes": {},
+					"children": [
+						{
+							"name": "children_1",
+							"type": "File",
+							"attributes": {
+								"custom_metric": 2
+							},
+							"children": []
+						}
+					],
+					"fixedPosition": {
+						"left": 5,
+						"top:": 5,
+						"width": 50,
+						"height": 30
+					}
+				},
+				{
+					"name": "folder_2",
+					"type": "Folder",
+					"attributes": {},
+					"children": [],
+					"fixedPosition": {
+						"left": 60,
+						"top:": 40,
+						"width": 35,
+						"height": 55
+					}
+				}
+			]
+		}
+	]
+}
+```
+
+Since a margin between folders is recommended, we set the coordinates for the folder in the top-left corner
+to `left: 5` and `top: 5`. The margin between the border of the map and `folder_1` is therefore `5`, but can be chosen at will.
+In order to set the coordinates to adjacent folders, we can again apply any margin between `folder_1` and `folder_2`. We chose
+a margin of `5` again, so that the coordinates of `folder_2` must be `left: 60` and `top: 40`.
+
+## Restrictions
+
+![Example]({{site.baseurl}}/assets/images/posts/how-to/fixate-folders/fixate-folder-example.jpg)
+
+In order to build a valid custom `.cc.json`, these rules must be followed:
+
+-   The values of `left`, `top`, `width` and `height` must be in range of `[0, 100]
+-   The value `left + width` or `top + height` must be in a range of `[0, 100]`.
+-   Folders may not overlap.
+-   Leaving space between folders (for visibility reasons) is recommended.
+-   Only folders on the root can have the `fixedPosition`-attribute.
+-   If there exists at least on fixated folder, all folders on the root must be fixated.
```

Comment:
```suggestion
-   All folders on the root level have to use the `fixedPosition` attribute.
```

Reply:
`on the root` still sounds misleading to me.
What about:
```suggestion
-   All direct sub-folders of the root have to use the `fixedPosition` attribute.
```

Reply:
```suggestion
-   All root folders have to use the `fixedPosition` attribute.
```

What about keeping it brief?

Reply:
There is only one root folder and it should not have the `fixedPosition` attribute. All direct children of that folder require it though, if at least one of them has it.

Reply:
```suggestion
-   All children of the root folder require the `fixedPosition` attribute.
```

How about that? (I explicitly did not mention "if at least one of them has the attribute set". This whole section is only about the fixed position)